lib/pbench/server/db/database.py

```diff
@@ -0,0 +1,54 @@
+import sys
+from sqlalchemy import create_engine
+from sqlalchemy.orm import scoped_session, sessionmaker
+from sqlalchemy.ext.declarative import declarative_base
+from pbench.server import NoOptionError, NoSectionError
+
+
+class Database:
+    """ Database - a base class for declaring SQLAlchemy database model
+    classes.
+    """
+
+    Base = declarative_base()
+    # Initialize the db scoped session
+    db_session = scoped_session(sessionmaker(autocommit=False, autoflush=False))
+
+    @staticmethod
+    def get_engine_uri(config, logger):
+        try:
+            psql = config.get("Postgres", "db_uri")
+            return psql
+        except (NoSectionError, NoOptionError):
+            if logger:
+                logger.error(
+                    "Failed to find an [Postgres] section" " in configuration file."
+                )
+            sys.exit(1)
+
+        # return f"postgresql+{psql_driver}://{psql_username}:{psql_password}@{psql_host}:{psql_port}/{psql_db}"
+
+    @staticmethod
+    def init_engine(server_config, logger):
+        try:
+            return create_engine(Database.get_engine_uri(server_config, logger))
+        except Exception:
+            if logger:
+                logger.exception("Exception while creating a sqlalchemy engine")
```

Comment:
`logger.exception` will implicitly dump the exception and stack trace in addition to the logger formatter including file and method, so the message is almost superfluous here.

Reply:
My point is that, while the logger will tell you where in the code the problem was detected, I'm concerned that that won't tell you what the problem actually was -- that is, if the create operation failed (and we'll know that in several ways),we won't know what the inputs to that operation were, and so how can we diagnose or rectify it without running a debugger?  And, more poignantly, if the root cause is something (missing or misconfigured) in the user's environment, it would be handiest if that were reported up-front, rather than forcing people to resort to looking at the code.

Reply:
Arguably on failure we could dump the entire set of config data because it might help in diagnosing any failure that occurs. At best, in most cases we can only guess at a root cause.

However in this case, I suppose the most likely problem is a bad `db_uri` and it wouldn't be that hard to add it...